src/spec/payid-easy-checkout.md

```diff
@@ -0,0 +1,378 @@
+yielding the URL `https://wallet.com/.well-known/webfinger?resource=payid%3Aalice%24wallet.com`. 
+The merchant UI would then [query the assembled URL](#step-2-query-payid-easy-checkout-discovery-url).
+
+The HTTP request in this example would look like this:
+    
+    GET /.well-known/webfinger?resource=payid%3Aalice%24wallet.com
+    Host: wallet.com
+    
+If the payer's PayID Discovery Server has enabled PayID Easy Checkout in their wallet, the server would respond with something like this:
+     
+     HTTP/1.1 200 OK
+     Access-Control-Allow-Origin: *
+     Content-Type: application/jrd+json
+
+     {
+       "subject" : "payid:alice$wallet.com",
+       "links" :
+       [
+         {  
+           "rel": "https://payid.org/ns/payid-easy-checkout-uri/1.0",
+           "template": "https://wallet.com/checkout"
+         }
+       ]
+     }
+
+### Assemble PayID Easy Checkout URL with Query Parameters
+The merchant UI would parse the PayID Discovery response and iterate over the 'links' collection to find the link with 
+the Relation Type of "https://payid.org/ns/payid-easy-checkout-uri/1.0". The merchant UI would then add all of the query
+parameters defined in [PayID Easy Checkout URL Query Parameters](#payid-easy-checkout-url-query-parameters) to the URL included in the JRD Link. 
+One query parameter of note is the "nextUrl" parameter, which allows the merchant to supply a redirect or callback URL 
+for the sender's wallet to call once the payer has confirmed the payment. In this example, the merchant would like 
+to display a "Thank You" page, and replaces `{nextUrl}` with `https://merchant.com/thankyou`.
+
+#### Correlating a Payment to an Invoice
+Merchants and non-profits will often times need to correlate discrete layer 1 payments to invoice or transaction entities
+in the merchants' native systems. The merchant in this example may have an invoice tracking system, on which an invoice
+gets created for the goods that the payer is buying with a unique identifier of `1045464`. A common practice for correlating
+layer 1 payments to a specific transaction or invoice is to accept payments on a different layer 1 address for each invoice
+so that the merchant can listen for payments into that address and tie the payment to the invoice.  However, because
+the PayID Easy Checkout URL only provides the receiver's PayID, there is currently no way to associate the address that
+is given to the payer to the invoice.
+
+In order to accomplish this, a merchant could provide a unique PayID containing the invoice identifier 
+for each PayID Easy Checkout transaction. In this example, the merchant would first associate a payment address with the
+invoice ID, and would then redirect the payer to their wallet with the `receiverPayId` query parameter set to `pay-1045464$merchant.com`.
+When the merchant PayID Server receives a query for the address associated with that PayID, they could return the previously
```

Comment:
I think the point here not really that there's something magical about the invoiceId being in the PayId itself -- the important part is that the merchant generates a unique L1 address and watches the L1 address for a payment to attach to a particular invoice. The fact that `1045464` is in the PayID doesn't add value to anyone other than allowing the PayID to be unique on a per-invoice basis.

Reply:
Maybe I should be more explicit about it.  I would say that the main point _is_ that there is some identifier embedded in the PayID, which the receiver's PayID server can parse out to correlate an invoice to a unique address they generate.